plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -0,0 +1,394 @@
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> namespace)
+    {
+        return ImmutableMap.of();
+    }
+
+    @Override
+    public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public List<SchemaTableName> listMaterializedViews(ConnectorSession session, Optional<String> namespace)
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "createMaterializedView is not supported for Iceberg REST catalog");
+    }
+
+    @Override
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "dropMaterializedView is not supported for Iceberg REST catalog");
+    }
+
+    @Override
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void renameMaterializedView(ConnectorSession session, SchemaTableName source, SchemaTableName target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "renameMaterializedView is not supported for Iceberg REST catalog");
+    }
+
+    @Override
+    public void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment)
+    {
+        loadTable(session, schemaTableName).updateSchema().updateColumnDoc(columnIdentity.getName(), comment.orElse(null)).commit();
+    }
+
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void updateViewComment(ConnectorSession session, SchemaTableName schemaViewName, Optional<String> comment)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "updateViewComment is not supported for Iceberg REST catalog");
+    }
+
+    private SessionCatalog.SessionContext convert(ConnectorSession session)
+    {
+        String sessionId = format("%s-%s", session.getUser(), session.getSource().orElse("default"));
+        String identity;
+
+        if (session.getIdentity().getPrincipal().isPresent()) {
+            identity = session.getIdentity().getPrincipal().get().getName();
+        }
+        else {
+            identity = session.getUser();
+        }
+
+        Map<String, String> properties = Map.of(
+                "identity", identity,
+                "user", session.getUser(),
```

Comment:
Setting the JWT token user to the trino session user will cause problems since most of the time the user that configured the oauth credential is *not* the trino session user. 

Should this instead be a separate config in `OAuth2SecurityConfig`? 

Example using a tabular REST catalog:

with the user explicitly set to the name of the credential creator:
```
 java -jar client/trino-cli/target/trino-cli-395-SNAPSHOT-executable.jar --user=redacted@redacted.com
trino> show schemas  in iceberg;
       Schema       
--------------------
 default            
 examples           
 information_schema 
 system             
(4 rows)
Query 20220929_213751_00003_pgu4h, FINISHED, 1 node
Splits: 11 total, 11 done (100.00%)
```

with the user not explicitly set:
```
Query 20220929_213639_00002_pgu4h failed: Not authorized: Unknown user: redacted
```

Reply:
@grantatspothero The user in the OAuth credential isn't required to match the session user.  This is just creating a JWT to pass along to the service for informational purposes and for the REST service to validate.  The REST service can choose to allow the session user to perform operations or to restrict or allow those operations. 

The credential set in the config authorizes Trino as a system, but then the REST service can perform identity federation knowing that Trino is a trusted service.

Reply:
To be clear, what is the suggested workaround for the above behavior (see the example I posted)?  

Right now when using a tabular catalog with iceberg, the trino session user has to be the same as the user who created the oauth token.

Reply:
This needs to be changed.  The principal is not a stable thing in Trino and is **only** there for logging.  Additionally, the user name is not reliable and should not be used.  Instead, let's just add a configuration property to set the value that should be pass through to rest interface.

Reply:
@dain Based on what I see available to the Catalog, the only information is either `session.getUser()` or `session.getIdentity()`.

Just want to clarify, what you're asking for is something like:

```
iceberg.metastore.rest.oauth2.identity = [ USER | PRINCIPAL ]
```

As a configuration?  I there are other identity options, just let me know and we can add them.

Reply:
no it should be `iceberg.metastore.rest.oauth2.identity = some@email.address` because that is what is required.  Otherwise this becomes a single user system

Reply:
Honestly, I expected the system to give us a single token that just worked.... now we are going to effectively have a token split over two configs... so it should effectively this is:
```
iceberg.metastore.rest.oauth2.token=some@emain.address::the-actual-token
```
Which I think will be pretty confusing to end users, but if that is how the system works I guess that is what we need to do 🤷

Reply:
I think we need to step back a second and clarify what's going on here, but I feel like there's a lot of confusion around how the OAuth2 credential/token is used and what is going on with the REST auth path and is getting wrapped up with the Tabular/Galaxy implementations.

The REST Catalog implementation follows a standard [OAuth credential flow](https://auth0.com/docs/get-started/authentication-and-authorization-flow/client-credentials-flow), which exchanges the credential for a token.  The catalog also supports sessionization through a [token exchange flow](https://www.rfc-editor.org/rfc/rfc8693.html) where a more specifically scoped token can optionally be provided.  These tokens are largely opaque to Trino as they're really a server side mechanism to manage authorization (as is the case with most OAuth2 tokens).

Currently, the session user/identity is the most relevant information available as to who is trying to access the underlying table resource.  This is somewhat equivalent to what Trino is doing for [trasactions in Hive](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java#L1410), [Hive privileges/grants](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SqlStandardAccessControlMetadata.java#L93), and various other places for ownership/access (glue, delta lake, etc.).  The server side implementation is responsible for how it wants to use access and information provided by Trino (it log/allow/deny based on a number of factors and is really an implementation detail of a server).

As @dain points out, the session user/identity isn't entirely reliable and can be problematic for more complicated cases (like views), but at this point it's the best/only information available and is generally consistent with other approaches until a better alternative is available.  We can always improve fidelity of the information and there are paths forward for handling things like views.

The other issues that @grantatspothero has been running into is more specifically related to Tabular/Galaxy integration and is just an artifact of how the OAuth2 enforcement is being handled.  We can work that out separately, but it really has no bearing on this particular PR.  Addressing these issues shouldn't require any changes in this particular PR.

Reply:
I think @danielcweeks approach makes a lot of sense. Using client credentials to establish machine to machine communication in OAuth2 authorization is pretty common and as such looks like a good solution for the job. Token exchange flow is fairly new and is not yet widely adopted but provides an even better solution: it allows for acting on behalf of a user, exchanging (in this case service's token) for a user token (effectively impersonating the user).

With regards to principal/user: is there anything better which could be used to identity the user in Trino? As I understand username is used only to identify which user issued the query in Trino and it's at the REST service discretion what it's going to do with it but it's not used for the authentication itself.

Reply:
Passing the user is impersonation, which has various implications. We need a config for this which is disabled by default. When disabled, we need a config to specify the shared system user to pass here.

If we're going to support impersonation, I think we need some guarantees in the REST catalog specification that will  not prevent caching:

* The user is only used for access control, not auditing.
* Name mappings and table data are the same for all users. In other words, `loadTable()` will return the same information for all users, or will be denied. It won't return different information depending on the user.